packages/models-library/src/models_library/api_schemas_webserver/users.py

```diff
@@ -0,0 +1,130 @@
+import re
+from datetime import date
+from enum import Enum
+from typing import Annotated, Literal
+
+from models_library.api_schemas_webserver.groups import MyGroupsGet
+from models_library.api_schemas_webserver.users_preferences import AggregatedPreferences
+from models_library.basic_types import IDStr
+from models_library.emails import LowerCaseEmailStr
+from models_library.users import FirstNameStr, LastNameStr, UserID
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+
+from ._base import InputSchema, OutputSchema
+
+
+class ProfilePrivacyGet(OutputSchema):
+    hide_fullname: bool
+    hide_email: bool
+
+
+class ProfilePrivacyUpdate(InputSchema):
+    hide_fullname: bool | None = None
+    hide_email: bool | None = None
+
+
+class ProfileGet(BaseModel):
+    # WARNING: do not use InputSchema until front-end is updated!
+    id: UserID
+    user_name: Annotated[
+        IDStr, Field(description="Unique username identifier", alias="userName")
+    ]
+    first_name: FirstNameStr | None = None
+    last_name: LastNameStr | None = None
+    login: LowerCaseEmailStr
+
+    role: Literal["ANONYMOUS", "GUEST", "USER", "TESTER", "PRODUCT_OWNER", "ADMIN"]
+    groups: MyGroupsGet | None = None
+    gravatar_id: Annotated[str | None, Field(deprecated=True)] = None
+
+    expiration_date: Annotated[
+        date | None,
+        Field(
+            description="If user has a trial account, it sets the expiration date, otherwise None",
+            alias="expirationDate",
+        ),
+    ] = None
+
+    privacy: ProfilePrivacyGet
+    preferences: AggregatedPreferences
+
+    model_config = ConfigDict(
+        # NOTE: old models have an hybrid between snake and camel cases!
+        # Should be unified at some point
+        populate_by_name=True,
+        json_schema_extra={
+            "examples": [
+                {
+                    "id": 42,
+                    "login": "bla@foo.com",
+                    "userName": "bla42",
+                    "role": "admin",  # pre
+                    "expirationDate": "2022-09-14",  # optional
+                    "preferences": {},
+                    "privacy": {"hide_fullname": 0, "hide_email": 1},
+                },
+            ]
+        },
+    )
+
+    @field_validator("role", mode="before")
+    @classmethod
+    def _to_upper_string(cls, v):
+        if isinstance(v, str):
+            return v.upper()
+        if isinstance(v, Enum):
+            return v.name.upper()
+        return v
+
+
+class ProfileUpdate(BaseModel):
+    # WARNING: do not use InputSchema until front-end is updated!
```

Comment:
should you not update it then? fullstack or not? 🤣

Reply:
I will do it in the coming PRs.